tests/unit/libutil-support/tests/nix_api_util.hh

```diff
@@ -33,5 +35,14 @@ protected:
         std::string msg(p, n);
         FAIL() << "nix_err_code(ctx) != NIX_OK, message: " << msg;
     }
+
+    inline void assert_ctx_err()
```

Comment:
Not immediately necessary, but it would be good to check the kind of error, and return the message (for when that's relevant to check).

```suggestion
    inline std::string assert_ctx_err(nix_err expected_err)
```

or `const char *`

Reply:
Agreed, but I'd prefer to open a new PR, this one is getting big enough.